components/Table/Table.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div>
+    <table>
+      <thead>
+        <tr>
+          <slot name="columns">
+            <th v-for="(head,i) in headers" :key="i">{{ head.text}}</th>
+            <th v-show="showExpand"></th>
+          </slot>
+        </tr>
+      </thead>
+      <tbody>
+        <template v-for="item in items">
+          <tr :key="item.link" @click="expandDetail(item.id)">
+            <slot :row="item">
+              <!-- default content -->
+              <td v-for="(header,i) in headers" :key="i">{{ item[header.value]}}</td>
+              <td v-show="showExpand">                
+                <i class="fas" :class="src(item.id)"></i>
+              </td>
+            </slot>
+          </tr>
+          <tr v-show="showExpand" v-if="toggle.includes(item.id)" :key="item.id">
+            <td :colspan="columnSpan">
+              <slot name="expandItem" :distributions="item.distributions"> 
```

Comment:
this component doesn't need to know about distribution what if I want to use it for a user list for example.

Reply:
Removed distribution. If you want to use it you can create user list content and into the slot.